utilities/backup/backup_engine.cc

```diff
@@ -1383,6 +1383,7 @@ IOStatus BackupEngineImpl::CreateNewBackupWithMetadata(
           uint64_t size_bytes = 0;
           IOStatus io_st;
           if (type == kTableFile || type == kBlobFile) {
+            // TODO: can we just use size_limit_bytes and remove this call?
```

Comment:
My guess is that due to [this feature](https://github.com/facebook/rocksdb/blob/7.2.fb/include/rocksdb/metadata.h#L66-L69), `size_limit` that passed in (e.g, [here](https://github.com/facebook/rocksdb/blob/main/utilities/checkpoint/checkpoint_impl.cc#L268)) can be smaller than the actual file size, which might not be what the proceeding steps want - @pdillinger might know more.

(By the way, I usually put confirmed TODO in code and put my "in-question" as a PR comment to reviewers, although I do see a couple of "in-question" TODO in the code base and I don't know why ....)

Reply:
Ah right, I remember asking @ajkr about this too, he mentioned that `size_limit` limits how many bytes to copy from WAL/MANIFEST. However, I thought SST files and Blob files are immutable so their size is fixed and not trimmed (as they are not "live" files)?

Reply:
Yes, `size_limit` may be smaller for mutable files (WAL and MANIFEST) because those files can continue to be appended after we have taken a snapshot of live files, but we do not want those newer appends in the checkpoint/backup. Otherwise the MANIFEST might reference SST files not included in the backup/checkpoint, for example.

For SST files and blob files we should not ever limit to a portion of the full file size. I did not know we even populate `size_limit` for these files until you told me, but since we do it needs to be >= the file size.

Reply:
I've removed the TODO comment for now and we can continue the discussion here or offline.